src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -653,17 +653,18 @@ trait TypeDiagnostics {
       unusedPrivates.traverse(body)
 
       if (settings.warnUnusedLocals || settings.warnUnusedPrivates) {
+        def shouldWarnOn(sym: Symbol) = if (sym.isPrivate) settings.warnUnusedPrivates else settings.warnUnusedLocals
         val valAdvice = "is never updated: consider using immutable val"
-        for (defn: DefTree <- unusedPrivates.unusedTerms) {
+        def termWarning(defn: SymTree): Unit = {
           val sym = defn.symbol
           val pos = (
             if (defn.pos.isDefined) defn.pos
             else if (sym.pos.isDefined) sym.pos
             else sym match {
               case sym: TermSymbol => sym.referenced.pos
-              case _               => NoPosition
+              case _ => NoPosition
```

Comment:
wait, you unaligned the arrows? Just kidding, in a way.

Reply:
I agree, NoPosition is in the wrong position.